src/System.Net.HttpListener/src/System/Net/HttpListener.Windows.cs

```diff
@@ -102,104 +71,69 @@ internal SafeHandle RequestQueueHandle
             }
         }
 
-        public AuthenticationSchemeSelector AuthenticationSchemeSelectorDelegate
+        private void ValidateV2Property()
```

Comment:
Diverging the common code considerably from other platforms will create issues in the long run (backporting will become harder).
What we did in other components is create an actual PAL layer containing common then several OS-specific files: HttpListenerPal.cs; HttpListenerPal.Windows.cs; HttpListenerPal.Unix.cs, etc.